### ISSUE TITLE
Compound fields don't have min/max occurs

### ISSUE THREAD

**Issue by tefra** [tefra/xsdata#748]
It seems that most restrictions are in the dataclass, but minOccurs and maxOccurs are not. See the example:

```
    <xs:simpleType name="astroPyPath">
        <xs:restriction base="xs:string">
            <xs:pattern value="hdul(\[.+\]).(data|header)(\[([\w-]+|[:\d,]+)\])+"/>
        </xs:restriction>
    </xs:simpleType>

    <xs:simpleType name="someNumber">
        <xs:restriction base="xs:float">
            <xs:minExclusive value="10"/>
            <xs:maxInclusive value="100"/>
        </xs:restriction>
    </xs:simpleType>

    <xs:complexType name="functions">
        <xs:choice>
            <xs:element name="Statistics" type="xs:string" minOccurs="4" maxOccurs="12"/>
            <xs:element name="Transforms" type="someNumber"/>
            <xs:element name="Mathematics" type="mathematicFunctions"/>
            <xs:element name="ReadFits" type="astroPyPath" minOccurs="1" maxOccurs="unbounded"/>
        </xs:choice>
    </xs:complexType>
```

becomes:

```
@dataclass
class Functions:
    class Meta:
        name = "functions"

    choice: List[object] = field(
        default_factory=list,
        metadata={
            "type": "Elements",
            "choices": (
                {
                    "name": "Statistics",
                    "type": str,
                    "namespace": "",
                },
                {
                    "name": "Transforms",
                    "type": float,
                    "namespace": "",
                    "min_exclusive": 10.0,
                    "max_inclusive": 100.0,
                },
                {
                    "name": "Mathematics",
                    "type": MathematicFunctions,
                    "namespace": "",
                },
                {
                    "name": "ReadFits",
                    "type": str,
                    "namespace": "",
                    "pattern": r"hdul(\[.+\]).(data|header)(\[([\w-]+|[:\d,]+)\])+",
                },
            ),
        }
    )
```

_Originally posted by @martinwiesmann in https://github.com/tefra/xsdata/issues/737#issuecomment-1387073943_

**Comment by tefra** [comment]
I am closing this one, for now it's not super important

**Comment by martinwiesmann** [comment]
Maybe not super important, but would be useful to have it fixed.
Would you mind reopening this issue, so that it doesn't get forgotten?

**Comment by martinwiesmann** [comment]
Thanks. And thank you for your great work.

**Comment by tefra** [comment]
This is now merged #994
